src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java

```diff
@@ -182,6 +212,55 @@ public ActiveDirectorySecurityRealm(String domain, String site, String bindName,
         }
 
         this.server = server;
+        this.cache = cache;
+
+        if (cache != null) {
+            userCache = userCacheBuilder
+                    .maximumSize(cache.getSize())
+                    .expireAfterWrite(cache.getTtl(), TimeUnit.SECONDS)
+                    .build();
+
+            groupCache = groupCacheBuilder
+                    .maximumSize(cache.getSize())
+                    .expireAfterWrite(cache.getTtl(), TimeUnit.SECONDS)
+                    .build();
+        } else {
+            userCache = userCacheBuilder
+                    .maximumSize(256)
+                    .expireAfterWrite(600, TimeUnit.SECONDS)
+                    .build();
+
+            groupCache = groupCacheBuilder
+                    .maximumSize(256)
+                    .expireAfterWrite(600, TimeUnit.SECONDS)
+                    .build();
+        }
+    }
+
+    @Restricted(NoExternalUse.class)
+    private CacheConfiguration getCache() {
+        return cache;
+    }
+
+    @Restricted(NoExternalUse.class)
+    private void setCache(CacheConfiguration cache) {
+        this.cache = cache;
+    }
+
```

Comment:
Are those methods needed when you have the field? Can the value change after construction? Can the field be `private` (if only filled through the `DataBoundConstructor`)?